### ISSUE TITLE
Slab back to full block

### ISSUE THREAD

**Issue by CodePhase** [minetest-mods/moreblocks#112]
It seems like building same material slabs on top of each other no longer converts them back to a full block.  This includes overridden native materials like wooden slabs made from wooden planks.

**Comment by fluxionary** [comment]
not sure what's going on here, but i intend to "fix" all the cruft that the `stairs` mod from `minetest_game` introduces via https://github.com/minetest-mods/moreblocks/pull/191

**Comment by fluxionary** [comment]
Oh, this means that if you place a slab on top of another slab, the two should combine into a single node? That's... interesting. I'm guessing that's not a bad idea, but it seems ... "tricky" to implement, as it depends on the orientation of the node (param2 value) and also which "face" a player is pointing at. Also, if you try to place a `slab_8` on top of a `slab_12`, should the result be a full node below, and a `slab_4` above? If you allow that sort of thing, what do you do if there's not free space "above"? Refuse to fill in the below at all? Place part of the node and return the rest to the player's inventory? What if there's no room remaining?

**Comment by S-S-X** [comment]
> Place part of the node and return the rest to the player's inventory? What if there's no room remaining?

I'd say no at least for this as player does not have usable circular saw to cut it.
Ability to stack sounds fine but cutting while stacking if it wont fit does not sound like correct behavior.
I think correct behavior would be to just prevent placement if there's not enough room for end result, it would also better follow common placement behavior with regular nodes.

There will still be some not so intuitive behaviors (assuming you're not developer and do not consider technical constraints) and possibly behavior that would not follow my opinion above when placing full size nodes on top of slabs.

**Comment by fluxionary** [comment]
> I'd say no at least for this as player does not have usable circular saw to cut it (result in world will and have to contain slab sizes player did not have but do not cut before placement and return different sizes). Ability to stack sounds fine but cutting while stacking if it wont fit does not sound like correct behavior. I think correct behavior would be to just prevent placement if there's not enough room for end result, it would also better follow common placement behavior with regular nodes.
> 
> There will still be some not so intuitive behaviors (assuming you're not developer and do not consider technical constraints) and possibly behavior that would not follow my opinion above when placing full size nodes on top of slabs (or just end result reaching next node above, different materials, nodes that can't be cut etc.).

my current feeling on this, is that it's a valid feature request, but i'm not going to tackle it in the context of #191

**Comment by S-S-X** [comment]
Yes very much agreed it is valid and should be very much possible to implement it without causing too much confusion.
